### ISSUE TITLE
Missing blog posts?

### ISSUE THREAD

**Issue by lbain** [middleman/middleman-blog#327]
Hello!
I'm working on upgrading to Middleman V4 (only one year late!). I had some initial problems with needing my files to have `.html` as part of the file name, but after that was sorted the main pages of my site were running fine.

However the blog posts are still not building. The index page for the blog builds just fine, but when I click through to look at a particular post I see an empty page. When I build my site the files for each blog post are present, but empty. I don't get any errors in the console, and when I add `puts` statements to the blog and main layouts they print without issue.

Most of my posts are `.html.md` though I have a few old `.html.haml` - neither load. I tried changing a post to just `.html` and then I could see the raw content of the post, neither converted from markdown (which makes sense as I took away the `.md`) nor with the main layout html applied.

My blog config options are:

```
activate :blog do |blog|
  blog.prefix = "blog"
  blog.permalink = "{year}/{title}.html"
  blog.sources = "{year}/{month}-{day}-{title}.html"
  blog.layout = "layouts/blog"
  blog.tag_template = "blog/tag.html"
  blog.paginate = false
end
```

You can find my site [here](https://github.com/lbain/lucybain/tree/middleman-v4) note that this is a branch - I haven't merged to master yet!

Any advice? What additional information would be helpful?

Thanks very much!
Lucy

**Comment by lbain** [comment]
Just a quick update from me: I didn't resolve this issue, but I was able to work around it. I believe I was experiencing the same issue as https://github.com/middleman/middleman-blog/issues/324 and https://github.com/middleman/middleman-blog/issues/322. Based on the insight from @corroded in https://github.com/middleman/middleman-blog/issues/322 I converted my layout files from `.haml` to `.erb` files. There were a few more changes required, but ultimately I believe that change was the piece I needed.

I will close this issue since https://github.com/middleman/middleman-blog/issues/322 is tracking the Haml layout issue already.
